src/commands/android/subcommands/common.ts

```diff
@@ -74,3 +76,138 @@ export function showMissingBinaryHelp(binaryName: SdkBinary) {
+  const configsPassed = optionsPassed.filter(option => !allowedFlagNames.includes(option));
+
+  // CHECK THE VALIDITY OF FLAG(s) PASSED
+
+  if (flagsPassed.length > 1) {
+    // A subcommand can only take one flag at a time.
+    Logger.log(`${colors.red(`Too many flags passed for '${subcommand}' subcommand:`)} ${flagsPassed.join(', ')} ${colors.gray('(only one expected)')}`);
+    showHelp(subcommand);
+
+    return false;
+  }
+
+  if (allowedFlags.length && flagsPassed.length === 0) {
+    // If the subcommand expects a flag but it is not passed:
+    // - if instead some other options are passed, throw error (we don't know if the options passed are configs and for which flag).
+    // - if no other options are passed, then we can prompt them for the flag and related configs.
+    if (configsPassed.length > 0) {
+      Logger.log(`${colors.red(`Unknown flag(s) passed for '${subcommand}' subcommand:`)} ${configsPassed.join(', ')}`);
+      showHelp(subcommand);
+
+      return false;
+    }
+
+    return {
+      subcommandFlag: '',
+      configs: []
+    };
+  }
+
+  // CHECK THE VALIDITY OF CONFIGS PASSED
+
+  const subcommandFlag = flagsPassed[0] || ''; // '' if no flag is allowed for the subcommand.
+
+  if (configsPassed.length === 0) {
+    // If no configs are passed, then we simply return and continue with the default subcommand flow.
+    return {
+      subcommandFlag,
+      configs: []
+    };
+  }
+
+  let allowedConfigs: CliConfig[] = [];
+  let configsFor = '';
+  if (!allowedFlags.length) {
+    allowedConfigs = AVAILABLE_SUBCOMMANDS[subcommand].cliConfigs || [];
+    configsFor = ` for '${subcommand}' subcommand`;
+  } else {
+    allowedConfigs = allowedFlags.find(flag => flag.name === subcommandFlag)?.cliConfigs || [];
+    configsFor = ` for '--${subcommandFlag}' flag`;
+  }
+
+  if (allowedConfigs.length) {
+    // Check if the passed configs are valid.
+    const configNames: string[] = allowedConfigs.map(config => config.name);
+
+    const configAliases: string[] = [];
+    allowedConfigs.forEach(config => configAliases.push(...config.alias));
+    configNames.push(...configAliases);
+
+    const unknownConfigs = configsPassed.filter(option => !configNames.includes(option));
+    if (unknownConfigs.length) {
+      Logger.log(`${colors.red(`Unknown config(s) passed${configsFor}:`)} ${unknownConfigs.join(', ')}`);
+      showHelp(subcommand);
+
+      return false;
+    }
+
+    // set main config in `options` if config aliases are passed.
+    const aliasToMainConfig: {[key: string]: string} = {};
+    allowedConfigs.forEach(config => {
+      config.alias.forEach(alias => {
+        aliasToMainConfig[alias] = config.name;
+      });
+    });
+
+    configsPassed.forEach((configName) => {
+      if (aliasToMainConfig[configName]) {
+        // `configName` is an alias
+        const mainConfig = aliasToMainConfig[configName];
+        options[mainConfig] = options[configName];
+      }
+    });
```

Comment:
In the case of configs, we would not use `minimist` for handling aliases but we'll handle them ourselves (to avoid polluting the `src/index.ts` file and get more control over aliases).

In the above block of code, if we encounter an alias set of `options`, we would set the main config name with the same value, so that we can rely on just the main config name in our scripts.
(Since we are not using `minimist` for aliases, the main config name will not be set automatically on `options`.)